### DISCUSSION TITLE
Auth with SSO and jwt tokens to rest-api

### DISCUSSION THREAD

**Discussion by IvanTurgenev1** [Budibase/budibase#5764] [Help]
Hello.
Please tell me if it is possible to implement the following scheme:
User logs in using sso keylock
and makes requests to the API using the token with which he was authorized in budibas?
i.e., the authorization that was used to login to budibase is used as the authentication header for accessing the rest api ?
thanks 
regards and best wishes

**Comment by melohagan** [comment]
Hello @wNvSp1r1T 

Is this for Google SSO? 

This discussion and issue may be relevant:

- https://github.com/Budibase/budibase/discussions/4218
- https://github.com/Budibase/budibase/issues/4262

**Reply by IvanTurgenev1** [reply]
Hello @melohagan 
Nope, i'm talk about keycloak SSO.
but, thanks very much for answer!

**Comment by IvanTurgenev1** [comment]
or maybe it's possible to pass UserName to header when we make api requests, it's gonna be enought

**Reply by melohagan** [reply]
Well you could pass the user email in the header of API requests:

- https://docs.budibase.com/docs/users-table#current-user-bindings
- https://docs.budibase.com/docs/rest-bindings

**Reply by IvanTurgenev1** [reply]
do I understand correctly? 
I can just pass this value in header like {{Current User.Email}}
or can i get example how to do it? many thanks and best regards

**Reply by IvanTurgenev1** [reply]
> Well you could pass the user email in the header of API requests:
> 
> * https://docs.budibase.com/docs/users-table#current-user-bindings
> * https://docs.budibase.com/docs/rest-bindings

and sorry for this stupid question. 
but how to pass values in forms to rest-api request? same as  {{Current User.Email}}? 
many thanks and best regards

**Reply by melohagan** [reply]
Hey @wNvSp1r1T 

Yes you would pass that binding through. So this section: https://docs.budibase.com/docs/rest-bindings#data-provider-applications-provider
shows how to pass it through, although in that example it's passing a form field. 

As well as data source queries, you can also use the **Execute Query** action.

**Comment by IvanTurgenev1** [comment]
@melohagan  thanks very much for your responce.
I have one last question about it.
I made request by button with action with "execute query"
it's work like a charm, but how i can access response data?
I'll just want to see response when press button, is it possibly? 
thanks

**Reply by IvanTurgenev1** [reply]
or i just need to store data in a row ? and after work with it? 
i was trying to do it, but i get smth like
{"data":[{"value":"ok"}],"pagination":{"cursor":null},"raw":"{\"health\":\"ok\"}","headers":{"date":"Tue, 10 May 2022 09:22:29 GMT","server":"uvicorn","content-length":"15","content-type":"application/json","set-cookie":"c8a73397166dfd75d068a14f2e59fef7=7d456f0d579353501ed4d2228d8a31fa; path=/; HttpOnly","cache-control":"private"}}
and i'm just need to print result like
"health": "ok"
and if we use data_row, how we can show only result for current user? 
i hope, it's just possibly to show response data :D

**Reply by melohagan** [reply]
There's a bit more on it here: https://docs.budibase.com/docs/referencing

If you use JavaScript then you can access the data as `$("Action X.Query Result").data[0]['value']`
